src/simplewallet/simplewallet.cpp

```diff
@@ -2295,6 +2299,42 @@ bool simple_wallet::stop_mining(const std::vector<std::string>& args)
   return true;
 }
 //----------------------------------------------------------------------------------------------------
+bool simple_wallet::set_node(const std::vector<std::string>& args)
+{
+  std::string node_url;
+
+  uint64_t below = 0;
+  if (args.size() < 1)
+  {
+    fail_msg_writer() << tr("missing node URL argument");
+    return true;
+  }
+
+  boost::regex rgx("^(.*://)?([A-Za-z0-9\\-\\.]+)(:[0-9]+)?");
+  boost::cmatch match;
+  // If user input matches URL regex
+  if (boost::regex_match(args[0].c_str(), match, rgx))
+  {
+    if (match.length() < 4)
+    {
+      LOG_ERROR("Unexpected array length - Exited simple_wallet::set_node()");
+      return true;
+    }
+    // If no port has been provided, use the default from config
+    if (!match[3].length())
+    {
+      int daemon_port = m_wallet->testnet() ? config::testnet::RPC_DEFAULT_PORT : config::RPC_DEFAULT_PORT;
+      node_url = match[1] + match[2] + std::string(":") + std::to_string(daemon_port);
+    } else {
+      node_url = args[0];
+    }
+    m_wallet->init(node_url);
```

Comment:
Need to call LOCK_IDLE_SCOPE here as the wallet might be refreshing in the background.

Reply:
fixed